commons/src/main/java/com/powsybl/commons/extensions/Extendable.java

```diff
@@ -49,4 +49,39 @@
      * @return all extensions associated to this extendable object.
      */
     <E extends Extension<O>> Collection<E> getExtensions();
+
+    /**
+     * Returns a name that is used to find matching {@link ExtensionAdderProvider}s
+     * when selecting implementations of extensions in {@link #newExtension}. This
+     * is meant to be overriden by extendables when multiple implementations exist.
+     *
+     * @return the name
+     */
+    default String getImplementationName() {
+        return "Default";
+    }
```

Comment:
It's strange to have this in `commons`. Is it specific to IIDM extensions?

Reply:
Not particularly. The getImplementationName method is only compared to the getImplementationName method of an ExtensionAdderProvider (when you call Extendable#newExtensionAdder). It's never compared to the NetworkFactoryServiceImpl from the iidm default implementation (although it happens to be the same constant "Default")

This is used when you have multiple implementations of an extendable, and you want the matching implementation of the extension. So this means that for all existing extendables today where we want multiple implementations, we can just create a Provider with the name "Default". But if we want we can also give the existing extendable an implementation name and create a Provider with this name.  For new extendable implementations we have to override and choose a new name (and use it in their providers).